File: 003_RECV.py
'''
My solution to Rosalind Bioinformatics Problem 003

Title: Complementing a Strand of DNA
Rosalind ID: RNA
Rosalind #: 003
URL: http://rosalind.info/problems/revc
'''

from Bio.Seq import Seq
import Bio.Alphabet

# open txt file with sequence to transcribe
f = open('data/rosalind_revc.txt', 'r')
# read all lines by removing newline character
data = f.read().replace('\n', '')
# close file
f.close()

# assign sequence as a DNA sequence
t = Seq(data, Bio.Alphabet.IUPAC.unambiguous_dna)
# use transcribe function
o = open('output/003_REVC.txt', 'w')
o.write(str(t.reverse_complement()))
o.close()

File: 003_RECV/003_RECV.R
if(FALSE){
  "
  My solution to Rosalind Bioinformatics Problem 003
  
  Title: Complementing a Strand of DNA
  Rosalind ID: 
  Rosalind #: 003
  URL: http://rosalind.info/problems/revc

  Goal to provide the reverse complement of a DNA string.
  "
}

require(Biostrings)
file_name = "data/rosalind_revc.txt"
raw_data = readChar(file_name, file.info(file_name)$size)

# remove return/newline characters
raw_seq = gsub("\r\n","",raw_data)
dna = DNAString(raw_seq)

# Reverse Complement
revc = as.character(reverseComplement(dna))

outhandle = "utput/003_REVC.txt"
# nchar - size of character string
writeChar(revc, outhandle, nchar(revc))